integration_tests/test_model_upload.py

```diff
@@ -30,10 +77,22 @@ def test_model_upload_and_versioning(self) -> None:
         model_upload(self.handle, self.temp_dir, LICENSE_NAME)
 
         # Create Version
-        model_upload(self.handle, self.temp_dir, LICENSE_NAME)
+        upload_with_retries(self.handle, self.temp_dir, LICENSE_NAME)
 
         # If delete model does not raise an error, then the upload was successful.
 
+    def test_model_upload_and_versioning_zip(self) -> None:
```

Comment:
Could you test that the upload is now respecting the hierarchy of a directory (subdirectories, etc...)?